internal/server/server.go

```diff
@@ -21,25 +24,44 @@ type Server struct {
 	cookies       *sessions.CookieStore
 	sessionName   string
 	sessionSecret string
+	apiServerURL  *url.URL
+	clusterCA     string
+	issuerURL     *url.URL
+	oauth2Config  *oauth2.Config
 }
 
-func New(opts ...func(*Server)) (http.Handler, error) {
-	randSecret, err := random.SecureString(32)
-	if err != nil {
-		return nil, err
-	}
+type ServerFuncOpt func(*Server) error
+
+func New(opts ...ServerFuncOpt) (http.Handler, error) {
 
 	// set defaults
 	s := &Server{
 		Router:        mux.NewRouter(),
 		template:      template.Must(template.ParseFS(embeddedFS, "templates/*.tmpl")),
 		sessionName:   "k8s-auth-portal-session",
-		sessionSecret: randSecret,
+		sessionSecret: "",
+		oauth2Config: &oauth2.Config{
+			ClientID:     "kubectl",
+			ClientSecret: "replace_this_public_client_secret",
```

Comment:
@dskatz We had discussed keeping this as random string as default, and pass in `"replace_this_public_client_secret"` as an argument, but then I realized `--kubectl-client-secret-filepath` requires a filepath, so we would've needed to put this string there, and then call the app with that arg for local testing.
It just seemed like a convoluted way to get the default ClientSecret, so I just left this as is. Let me know if you would me to change it.

Reply:
I think setting to random is a good default.   Deployment needs to be managed separately anyway.